doc/debuginfo.txt

```diff
@@ -0,0 +1,187 @@
+name, however, the analysis will give as a result the name of the
+nearest function symbol, plus an offset in bytes to the location in
+the executable from which the event originated.
+
+If the executable has neither ELF symbols nor DWARF information,
+babeltrace will be unable to map an event to its source location, and
+will simply display the instruction pointer (address), as in prior
+versions of babeltrace.
+
+Getting the Right Tracer
+------------------------
+
+Debug info analysis is performed automatically by babeltrace, provided
+the trace contains sufficient information. In order to be able to
+trace all the necessary information, the following software is
+required:
+
+    * lttng-ust version 2.8.0-pre or later
+    * lttng-tools, corresponding version
+
+LTTng 2.8 is currently in development, which is why the pre-release
+version is needed to have access to the latest features allowing debug
+info analysis in babeltrace.
+
+You can get these from source at:
+
+    * https://github.com/lttng/lttng-ust
+    * https://github.com/lttng/lttng-tools
+
+Ubuntu users also have the option of installing via the LTTng daily
+PPA:
+
+    * https://launchpad.net/~lttng/+archive/ubuntu/daily
+
+Tracing for Debug Info Analysis
+-------------------------------
+
+Babeltrace needs some extra information from contexts, namely ip and
+vpid, to perform its analysis. These can be enabled after the creation
+of a tracing session as follows:
+
+   $ lttng add-context -u -t ip -t vpid
+
+The tracing can now be performed as it normally would. Once the trace
+is collected, it can the be read by babeltrace for analysis.
+
+Analysing the Trace
+-------------------
+
+To perform the analysis, the trace can simply be read as it normally
+would:
+
+   $ babeltrace <path/to/trace>
+
+Debug info analysis is on by default, and will automatically print the
+extra source location information if it can find it. A sample output
+may look like this:
+
+    [...]
+    [19:53:41.648394410] (+0.000022616) colossus my_provider:my_first_tracepoint: { cpu_id = 2 }, { ip = 0x7F75AEDFB4D4, debug_info = { func = "foo", source_loc = "/home/efficios/example/libhello.c:7" }, vpid = 25577 }, { my_string_field = "hello, tracer", my_integer_field = 42 }
+    [19:53:41.648423786] (+0.000029376) colossus my_provider:my_first_tracepoint: { cpu_id = 2 }, { ip = 0x7F75AEDFB5A6, debug_info = { func = "bar", source_loc = "/home/efficios/example/libhello.c:13" }, vpid = 25577 }, { my_string_field = "recoltes et semailles", my_integer_field = 57 }
+    [19:53:41.648435726] (+0.000011940) colossus my_provider:my_other_tracepoint: { cpu_id = 2 }, { ip = 0x7F75AEDFB66B, debug_info = { func = "baz", source_loc = "/home/efficios/example/libhello.c:20" }, vpid = 25577 }, { some_field = 1729 }
+    [...]
+
+The interesting part is the debug_info section of the context:
+
+    debug_info = { func = "foo", source_loc = "/home/efficios/example/libhello.c:7" }
```

Comment:
I don't care about global boolean options that apply to a specific plugin, it's easy to handle in BT 2.0's config "translator". If it's in 1.4, I would allow it in 2.0 too.